src/main/java/net/javadiscord/javabot/listener/CodeAttachmentListener.java

```diff
@@ -0,0 +1,123 @@
+package net.javadiscord.javabot.listener;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Listens for messages containing "message.txt" or similar file attachments
+ * containing source code, in order to format the code into one or more code
+ * snippet blocks in the Discord channel instead. The code blocks are sent in
+ * a series of messages that chain their replies to the original message.
+ */
+public class CodeAttachmentListener extends ListenerAdapter {
+	@Override
+	public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+		if (isCodeMessage(event.getMessage())) {
+			fetchSource(event.getMessage())
+				.thenApply(this::processSource)
+				.thenAcceptAsync(blocks -> {
+					Message lastMsg = event.getMessage();
+					for (var block : blocks) {
+						lastMsg = lastMsg.reply(block).complete();
```

Comment:
Add `allowed_mentions` so that this cannot be exploited to ping `@everyone` or similar.

Reply:
Not necessarily needed, as it is disabled by default

Reply:
Mentions to `@everyone` are disabled by default but what about role mentions or mass pings of individual users?

Reply:
Yeah, true

Reply:
Added a thing to set allowed mentions to an empty set.